File: week9/day43/p1/AuthService/Models/UserModel.cs
using System.ComponentModel.DataAnnotations;

namespace AuthService.Models
{
    public class UserModel
    {
        [Key]
        public int UserId { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string Role { get; set; } //Admin ,User
    }

    public class AuthDto
    {
        public string Email { get; set; }
        public string Password { get; set; }
        public string Role { get; set; } // Only needed for registration
    }
}

File: B3_Angular_Korukonda_Syam_SaravanaKumar_ShopEZ – E-Commerce Application/Angular/src/app/models/user.model.ts
// Matches UserResponseDto from AuthService
export interface User {
  userId: number;
  userName: string;
  emailAddress: string;
  role: string;
}

// Matches LoginDto from AuthService
export interface LoginRequest {
  email: string;
  password: string;
}

// Matches RegisterDto from AuthService
export interface RegisterRequest {
  name: string;
  email: string;
  password: string;
  role: string;
}

// Matches AuthResponseDto from AuthService
export interface AuthResponse {
  token: string;
  role: string;
  message: string;
}
